src/addresses.rs

```diff
@@ -4,9 +4,49 @@ use mimirsbrunn::admin_geofinder::AdminGeoFinder;
 use mimirsbrunn::labels::format_addr_name_and_label;
 use mimirsbrunn::labels::format_street_label;
 use mimirsbrunn::utils::find_country_codes;
+use serde::Deserialize;
 use std::ops::Deref;
 use std::sync::Arc;
 
+/// Check if a mimir address originates from OSM data.
+pub fn is_osm_addr(addr: &mimir::Address) -> bool {
+    match addr {
+        mimir::Address::Addr(addr) => addr.id.contains(":osm:"),
+        mimir::Address::Street(st) => st.id.contains(":osm:"),
+    }
+}
+
+/// Get current value of address associated with a POI in the ES database if
+/// any, together with current coordinates of the POI that have been used to
+/// perform a reverse
+pub fn get_current_addr(
+    rubber: &mut Rubber,
+    poi_index: &str,
+    osm_id: &str,
+) -> Option<(mimir::Coord, mimir::Address)> {
+    let query = format!(
+        "{}/poi/{}/_source?_source_include=address,coord",
+        poi_index, osm_id
+    );
+
+    #[derive(Deserialize)]
+    struct FetchPOI {
+        coord: mimir::Coord,
+        address: mimir::Address,
+    }
+
+    rubber
+        .get(&query)
+        .map_err(|err| warn!("failed to connect to ES: {:?}", err))
+        .ok()
+        .and_then(|mut res| {
+            res.json()
+                .map_err(|err| warn!("failed to parse ES response: {:?}", err))
+                .ok()
+        })
+        .map(|poi_json: FetchPOI| (poi_json.coord, poi_json.address))
```

Comment:
How is an unknown POI (404 response) handled here ?
A `warn!` would not be very appropriate, as this is expected for all new POIs.

Reply:
That would indeed be an irrelevant spam of the logs, it should be fine now: https://github.com/QwantResearch/fafnir/pull/47/files#diff-9b894eb37fb1dd6f240a692c6167c7dfR42-R45

Also, do you think that there would be a clean way to remove these messages:
```
Apr 10 17:29:30.541 INFO Doing post_body_op on /munin_addr,munin_street/_search?ignore_unavailable=true
```
They appear to be raised here: https://github.com/benashford/rs-es/blob/master/src/lib.rs#L152. A line of logs for each request seems to be a bit to much? :/

Reply:
I think this message comes from `rs-es`.  
In [our deployed pipeline](https://github.com/QwantResearch/docker_mimir/blob/master/docker-compose.yml#L44) we are using `RUST_LOG=warn,fafnir=info,mimir=info` to hide verbose messages from dependencies.